BasicSample/Assets/GrabAndThrow/Scripts/GrabAndThrowBehavior.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR;
+using UnityEngine.XR.OpenXR.Features.Interactions;
+
+public class GrabAndThrowBehavior : MonoBehaviour
+{
+    private bool[] m_wasTriggerPressed = { false, false };
+    private const float SPEED_MULTIPLIER = 10;
+
+    // Update is called once per frame
+    private void Update()
+    {
+        // get trigger and velocity data from the left device
+        for (int i = 0; i < 2; i++)
+        {
+            InputDevice device = InputDevices.GetDeviceAtXRNode((i == 0) ? XRNode.RightHand : XRNode.LeftHand);
+            bool deviceHasData = device.TryGetFeatureValue(CommonUsages.isTracked, out bool deviceIsTracked);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.primaryButton, out bool isdeviceTapped);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceVelocity, out Vector3 deviceVelocity);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.devicePosition, out Vector3 devicePosition);
+            deviceHasData &= device.TryGetFeatureValue(CommonUsages.deviceRotation, out Quaternion deviceRotation);
+
+            if (deviceHasData && deviceIsTracked)
+            {
+                if (m_wasTriggerPressed[i] && !isdeviceTapped)
+                {
+                    // release the cube using velocity
+                    GetComponent<Rigidbody>().velocity = deviceVelocity * SPEED_MULTIPLIER;
```

Comment:
The velocity/angular velocity should be applied on the point where user grabs the rigidbody at devicePosition, instead applying them to the center of mass of the rigid body.

Reply:
Unity does not provide a way to apply velocity to a point on the rigid body, instead there is a way to AddForceAtPosition but I am not sure it is going to be the same effect. Also, I observe that i use far interaction when picking the cube up instead of near interaction. Is this something that I need to consider when applying force? It would be great if someone could test this and let me know what exactly I would want to achieve with this change.